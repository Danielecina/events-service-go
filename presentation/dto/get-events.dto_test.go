package dto

import (
	"encoding/json"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestGetEventsResponse(t *testing.T) {
	t.Run("Expect to serialize GetEventsResponse correctly", func(t *testing.T) {
		resp := []GetEventResponse{
			{
				Name:        "Evento",
				Description: "Descrizione",
				Location:    "Luogo",
				UserID:      "user-1",
			},
		}
		marshalResp, err := json.Marshal(resp)
		require.NoError(t, err)
		snaps.MatchSnapshot(t, string(marshalResp), "get_event_response.json")
	})
}

func TestGetEventsErrorResponse(t *testing.T) {
	t.Run("Expect to serialize GetEventsErrorResponse correctly", func(t *testing.T) {
		errResp := GetEventsErrorResponse{
			Message:    "errore",
			ErrorCode:  "E002",
			StatusCode: 404,
			Errors:     []string{"evento non trovato"},
		}
		marshalInput, err := json.Marshal(errResp)
		require.NoError(t, err)
		snaps.MatchSnapshot(t, string(marshalInput), "get_event_error_response.json")
	})
}
