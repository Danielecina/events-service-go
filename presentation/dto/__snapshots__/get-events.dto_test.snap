
[TestGetEventsResponse/Expect_to_serialize_GetEventsResponse_correctly - 1]
{"name":"Evento","description":"Descrizione","location":"Luogo","user_id":"user-1"}
get_event_response.json
---

[TestGetEventsErrorResponse/Expect_to_serialize_GetEventsErrorResponse_correctly - 1]
{"message":"errore","error_code":"E002","status_code":404,"errors":["evento non trovato"]}
get_event_error_response.json
---
